test/test_torch.py

```diff
@@ -3112,15 +3111,26 @@ def test_from_file(self):
                 t2.fill_(rnum)
                 self.assertEqual(t1, t2, atol=0, rtol=0)
 
-        @unittest.skipIf(IS_WINDOWS, "TODO: need to fix this test case for Windows")
+                # release the tensors
+                del s1, t1, s2, t2
+
+            with TemporaryFileName() as fname:
+                assert_with_filename(fname)
+
+            temp_dir = os.path.join(tempfile.gettempdir(), '中文')
+            os.mkdir(temp_dir)
```

Comment:
Is it better to create a context manager for this? Otherwise, if the test fail, then we will stop at line 3123. The cleanup won't be executed since it is at line 3124.

Reply:
Yeah I'm only testing this for CI. It drove me crazy. The `tempfile.TempraoryDirectory` also gets cleaned up prematurely. And there's no `delete=False`. So I have to do this manually.

Reply:
Do you have the Linux environment locally? If not, maybe you could try creating a docker container or using WSL and make sure things are working locally. Testing the whole thing through CI is way too slow.

Reply:
I have a working Linux machine but the environment is not the same as CI. I will give docker a try later.